### PR TITLE
Fixed Fertilities and Slots

### PR DESCRIPTION
**Pull request by Cyclonit**

### CODE REVIEW

**Comment by Shad0wlife** [comment]
This looks like a lot of stuff. I don't know how much of a review I can do here. I'll have to play around with the features though to see if any issues or errors crop up.

File: AnnoMapEditor/UI/Overlays/SelectSlots/SelectSlotsOverlay.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            if (e.OldValue is SelectSlotsViewModel oldViewModel)
+            {
+                oldViewModel.SlotAssignmentViewModels.CollectionChanged -= SlotAssignmentViewModels_CollectionChanged;
+                foreach (SlotAssignmentViewModel slotAssignment in oldViewModel.SlotAssignmentViewModels)
+                    slotAssignment.PropertyChanged -= SlotAssginment_PropertyChanged;
+            }
+
+            if (e.NewValue is SelectSlotsViewModel newViewModel)
+            {
+                newViewModel.SlotAssignmentViewModels.CollectionChanged += SlotAssignmentViewModels_CollectionChanged;
```
[the remark this answers is not in the document]
Comment (Shad0wlife) [reply AnnoMapEditor/UI/Overlays/SelectSlots/SelectSlotsOverlay.xaml.cs:70]: Fix found in https://github.com/Cyclonit/AnnoMapEditor/pull/3

**Review by Shad0wlife — CHANGES_REQUESTED** [review]
Very sophisticated island slot designer! Works as intended as far as I can tell (with the addition of the PR I added to this PR ;) )

File: AnnoMapEditor/DataArchives/Assets/Models/StandardAsset.cs [hunk 2]
```diff
@@ -18,8 +15,20 @@ public class StandardAsset : ObservableBase
...
+        }
+        private ImageSource? _icon;
+
+
+        protected StandardAsset()
+        {
+
+        }
 
         public StandardAsset(XElement valuesXml)
```
[the remark this answers is not in the document]
Comment (Cyclonit) [reply AnnoMapEditor/DataArchives/Assets/Models/StandardAsset.cs:33]: Good question... I think there was a reason to not make it abstract. But I cannot find it any more. I believe we can safely make it abstract.

File: AnnoMapEditor/DataArchives/Assets/Repositories/FixedIslandRepository.cs [hunk 3]
```diff
@@ -39,14 +39,89 @@ protected override Task DoLoad()
...
+
+                // get the island's size in tiles
+                Attrib sizeInTilesAttrib = infoDoc.GetAttrib("MapSize");
+                int sizeInTiles = BitConverter.ToInt32(sizeInTilesAttrib.Content, 0);
+
+                // get slots
+                // The list of slots is found at <ObjectMetaInfo><SlotObjects> within the
+                // a7mfile. It has two kinds of elements.
+                //
+                // First there are lists with a single item of the following form. They denote the
```
[the remark this answers is not in the document]
Comment (Cyclonit) [reply AnnoMapEditor/DataArchives/Assets/Repositories/FixedIslandRepository.cs:64]: Gogogo ;)

File: AnnoMapEditor/UI/Overlays/SelectSlots/SelectSlotsOverlay.xaml.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Cyclonit) [reply AnnoMapEditor/UI/Overlays/SelectSlots/SelectSlotsOverlay.xaml.cs]: Merged the PR.

File: AnnoMapEditor/DataArchives/Assets/Models/StandardAsset.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Shad0wlife) [reply AnnoMapEditor/DataArchives/Assets/Models/StandardAsset.cs:33]: Changed.

**Pull request merged**